tortoise/verifying_tortoise.go

```diff
@@ -395,6 +399,18 @@ func (t *turtle) processBlock(block *types.Block) error {
 	// TODO: save and vote against blocks that exceed the max exception list size (DoS prevention)
 	opinion := make(map[types.BlockID]vec)
 	for blk, vote := range baseBlockOpinion.BlocksOpinion {
+		fblk, err := t.bdp.GetBlock(blk)
+		if err != nil {
+			return fmt.Errorf("voted block not in db")
```

Comment:
add block ID to errmsg?

Reply:
I will be cooll to have in error messages layer id also.

Reply:
@lrettig - it will probably be useful for these cases if we could create a structure with a few fields and return it, right now it will be returned in string form..

@sudachen - well, it is problematic, since we don't really know the LayerID of this block since we couldn't find it in our DB.. we could print the original processed block id and layerid as well if that helps..

Reply:
@y0sher re: the block's layer - if you print the voting block's layer, then in most cases the voted-for block will be in the previous layer. This is not always the case, but can help the investigation.

Having an error structure with fields shouldn't be terribly hard. We can create a type that implements the `error` interface and pass that along, then in `log.Err()` we can do a type assertion and extract the fields if relevant. Only problem is that `log.Err()` returns a single field - would be nicer if we could log those extracted fields separately. I have an idea how to do that, but it starts getting complicated... If we make sure we always use `fmt.Errorf("bla: %w", err)` when propagating errors then we won't lose the wrapped fields either.

Reply:
we need a solution to pass multiple fields as error.. for sure not a part of this PR, would appreciate if you could create an issue about it and mention this code, we can outsource it to the community.

Reply:
Created issue here: https://github.com/spacemeshos/go-spacemesh/issues/2456

Reply:
thanks